src/test/shell/bazel/cpp_darwin_integration_test.sh

```diff
@@ -147,10 +149,61 @@ cc_library(
   srcs = ["foo.cc"],
   hdrs = ["foo.h"],
 )
+cc_shared_library(
+  name = "foo_shared",
+  deps = [":foo"],
+)
+cc_test(
+  name = "test",
+  srcs = ["test.cc"],
+  deps = [":foo"],
+  dynamic_deps = [":foo_shared"],
+)
+EOF
+  cat > cpp/foo.h <<EOF
+  int foo();
+EOF
+  cat > cpp/foo.cc <<EOF
+  int foo() { return 0; }
+EOF
+  cat > cpp/test.cc <<EOF
+  #include "cpp/foo.h"
+  int main() {
+    return foo();
+  }
+EOF
+
+  bazel test --incompatible_macos_set_install_name //cpp:test || \
+      fail "bazel test //cpp:test failed"
+  # Ensure @rpath is correctly set in the binary.
+  ./bazel-bin/cpp/test || \
+      fail "//cpp:test workspace execution failed, expected return 0, got $?"
+  cd bazel-bin
+  ./cpp/test || \
+      fail "//cpp:test execution failed, expected 0, but $?"
+}
+
+function test_cc_test_with_explicit_install_name_apple_support() {
+  cat > MODULE.bazel <<EOF
+bazel_dep(name = "apple_support", version = "1.16.0")
+EOF
```

Comment:
This is the only difference with `test_cc_test_with_explicit_install_name`. Does it make sense to have both tests, or should they just be merged into one?

Reply:
Merging them would require modifying the module file and running a full `clean --expunge` to realistically test the switch between the two toolchains. I think that this wouldn't be better.

I could extract a helper function for the setup, would you prefer that?